arangod/VocBase/Methods/Collections.cpp

```diff
@@ -263,14 +264,38 @@ Result Collections::create(TRI_vocbase_t& vocbase,
                info.collectionType != TRI_col_type_e::TRI_COL_TYPE_EDGE) {
       return TRI_ERROR_ARANGO_COLLECTION_TYPE_INVALID;
     }
+
     TRI_ASSERT(info.properties.isObject());
     helper.clear();
     helper.openObject();
     helper.add(arangodb::StaticStrings::DataSourceType,
                arangodb::velocypack::Value(static_cast<int>(info.collectionType)));
     helper.add(arangodb::StaticStrings::DataSourceName,
                arangodb::velocypack::Value(info.name));
+
+
+    auto replicationFactorSlice = info.properties.get(StaticStrings::ReplicationFactor);
+    if(replicationFactorSlice.isNone()) {
+      auto factor = vocbase.replicationFactor();
+      if(factor > 0 && vocbase.IsSystemName(info.name)) {
+        auto* cl = application_features::ApplicationServer::lookupFeature<ClusterFeature>("Cluster");
+        factor = std::max(vocbase.replicationFactor(), cl->systemReplicationFactor());
```

Comment:
```suggestion
        factor = std::max(factor, cl->systemReplicationFactor());
```
Not sure if `std::max(..., ...)` is right here. If the user has explicitly specified a replicationFactor template value for the database, why max it with the instance's systemReplicationFactor?

Reply:
Thought it might good to make sure that system collections have at least the replication factor that was set on the command-line.

Reply:
The code OK imo. In the general case the factor is initialized to `vocbase.replicationFactor()` if not `replicationFactor` is provided. In case we have a system collection, then we take the max of the replication factor set as default in the vocbase and the replication factor for system collections provided on the command-line. In case the user sets 1 as default replication factor for the database, we need this to ensure that we will have our standard of 2 copies for system collections. When somebody is paranoid and sets a high replication factor for the database to prevent data-loss, then he would be probably be surprised to see that important system collections have a lower replication factor than other collections in his database.
Please let me know how you want to change the behavior.